tests/test_joint_logprob.py

```diff
@@ -211,6 +211,7 @@ def test_joint_logprob_subtensor():
         np.testing.assert_almost_equal(logp_vals, exp_obs_logps, decimal=decimals)
 
 
+@pytest.mark.skip(reason="Disabled until we decide how we want this to work.")
```

Comment:
What is not clear here?

Reply:
It's not clear that we should guarantee the (object) identities of the value graphs bound to a measurable term.  It definitely seems like a convenient thing to have in some situations, but it shouldn't be necessary for any functionality.

Reply:
I think it would be confusing to not guarantee this identity for values that are passed as realizations of a measurable term, as is the case here. What if we observe not `Y_rv` directly, but another quantity from which its value can be derived like in the following example?

```python
import aesara
import aesara.tensor as at
from aeppl import joint_logprob


x = at.scalar("x")
beta_rv = at.random.normal(0, 1, name="beta")
Y_rv = at.random.normal(beta_rv * x, 1, name="y")

obs_at = Y_rv.clone()
y_vv = at.log(obs_at + 1.)
logp, (beta_vv,) = joint_logprob(beta_rv, realized={Y_rv: y_vv})
```

Shouldn't we be able to compile this function:

```python
logp_fn = aesara.function([obs_at, beta_vv], logp)
```

Reply:
> Shouldn't we be able to compile this function?

Yes, and for that example, I think all that's missing is `x` as an input.

Reply:
The main issue/question behind this one is that realized value-variable graphs like `y * 2` can have canonicalization applied to them, so that the log-probability output graph(s) will not contain the original `y * 2` subgraphs, but canonicalized ones like `2 * y`.  

This isn't actually an issue or limitation of any sort, especially since a user-held "original" value-variable graph can/will be merged with its canonicalized form when both graphs are put together at some point during compilation.  I recall using the old identity guarantee a long time ago for something, but that should no longer be necessary (and it never really should have been).

Anyway, as long as the identities of **inputs** to such graphs are preserved (i.e. `y` in our example), a user should be able to compile everything as expected.

I've updated the test to reflect these sentiments.